### DISCUSSION TITLE
Current State of Surface and Linux - Battery Life?

### DISCUSSION THREAD

**Discussion by ghost** [linux-surface/linux-surface#1148] [Q&A]
Hello everyone,

I'm a owner of a Surface Laptop Studio and must say that I really love Linux on my device.
However, the situation regarding Battery Life concerns me a bit so I would like to know potential ways to improve?
I already got my hands on `auto-cpufreq` but don't know if that's enough or if I should do more than that.
I've also `envycontrol` installed to turn the dGPU (Nvidia) off.

And then we have TLP... I know that there's the additional note which says that it shouldn't be used.
But I also found [this Reddit thread](https://www.reddit.com/r/SurfaceLinux/comments/esd5lr/should_you_use_tlppowertop/) here where one of the developers (?) explains that this part can be changed because he uses TLP too.
The thread is now almost 3 years old. Did something changed during the 3 years? And if yes, what do users recommend here for settings if they're using it?
Thanks in advance!

**Comment by RichardODonoghue** [comment]
@C0deViewer with auto-cpufreq did you enable the auto governor as a service? I've come across a few people who have installed it and then not set it up correctly and therefore have not gotten the results they desire.

**Reply by ghost** [reply]
@RichardODonoghue I believe I installed the daemon correctly and enabled it as a service. But this doesn't matter anyway because after the days where I tried Linux on my SLS I came to the conclusion that Linux is maybe ready to be used on normal Laptops, but not on 2-in-1 devices like the Surface series and the others (HP, Dell, etc.) and switched back to Windows. I came across several things which are in my opinion worth to note from my Linux experience: constant spinning fans even if I do basically nothing, Gnome has bad fractional scaling options (either too tiny for me at 100% or too big at 200% or bigger), hanging pen input (so it thinks I'm still 'holding' something on my screen somewhere even if I don't touch the screen at all), KDE has working fractional scaling options but a broken on-screen keyboard - alternatives don't work too), etc.

I know the mentioned things are more smaller things (more or less), but I need a device which I can trust on because it should be used for productivity work. So I need the device to work w/o any problems really. I can imagine using it more to make tests around with it but that's not a device I'm looking for right now. I hope that I can use Linux one day completely w/o having the fear that something could break. I appreciate the work which has been already done. I mean if I would try it on some other 2-in-1 laptops from any other company then it's probably gambling more or less because it's unknown how the compatibility is with pen, display mode recognition, etc.

So yeah, I wish the team behind the project the best that they can achieve the ultimate working Linux kernel for Surface devices. But there's still work to do behind other software parts like the DEs (Gnome and KDE).

PS: There are also hardware specific parts which makes me concern regarding the longevity of the device itself (little creaking sounds on the main hinge and bigger (I guess?) creaking sounds on the fold mechanism of the display). I've time to return it so... let's see how it holds for the next days.

Thanks for trying to help me anyway.

**Comment by Makronukleus** [comment]
Hi. I've been using ArchLinux on my SP7 for a couple of months now. I second the question about TLP, what's the state? And referring to the README - how to configure it properly?

**Reply by RichardODonoghue** [reply]
I can't speak for the SP7 but I have a SL4. One thing I always do when I install Linux on a laptop is install auto-cpufreq and turn the daemon on. It's the only way I've found to get decent battery life on a laptop with Linux without any compromises. TLP is garbage.
